compiler/rustc_monomorphize/src/partitioning.rs

```diff
@@ -187,7 +187,13 @@ where
     }
 
     // Ensure CGUs are sorted by name, so that we get deterministic results.
-    assert!(codegen_units.is_sorted_by(|a, b| Some(a.name().as_str().cmp(b.name().as_str()))));
+    if !codegen_units.is_sorted_by(|a, b| Some(a.name().as_str().cmp(b.name().as_str()))) {
+        let mut names = String::new();
+        for cgu in codegen_units.iter() {
+            names += &format!("- {}\n", cgu.name());
+        }
+        panic!("unsorted CGUs:\n{names}");
```

Comment:
maybe

```suggestion
        bug!("unsorted CGUs:\n{names}");
```

for the nicer error message if it somehow happens to others (outside of fuchsia people, who will be expecting it)

in any case r=me

Reply:
:+1: